bin/stress-test/src/metrics.rs

```diff
@@ -0,0 +1,176 @@
+use std::{
+    fmt::Display,
+    path::{Path, PathBuf},
+    process::Command,
+    time::Duration,
+};
+
+const SQLITE_TABLES: [&str; 11] = [
+    "account_deltas",
+    "block_headers",
+    "account_fungible_asset_deltas",
+    "notes",
+    "account_non_fungible_asset_updates",
+    "nullifiers",
+    "account_storage_map_updates",
+    "settings",
+    "account_storage_slot_updates",
+    "transactions",
+    "accounts",
+];
+
+/// Metrics struct to show the results of the stress test
+pub struct Metrics {
+    insertion_time_per_block: Vec<Duration>,
+    get_block_inputs_time_per_block: Vec<Duration>,
+    get_batch_inputs_time_per_block: Vec<Duration>,
+    bytes_per_block: Vec<usize>,
+    num_insertions: u32,
+    store_file_sizes: Vec<u64>,
+    initial_store_size: u64,
+    store_file: PathBuf,
+}
+
+impl Metrics {
+    /// Create a new Metrics instance.
+    pub fn new(store_file: PathBuf) -> Self {
+        let initial_store_size = Self::get_store_size(&store_file);
+        Self {
+            insertion_time_per_block: Vec::new(),
+            get_block_inputs_time_per_block: Vec::new(),
+            get_batch_inputs_time_per_block: Vec::new(),
+            bytes_per_block: Vec::new(),
+            num_insertions: 0,
+            store_file_sizes: Vec::new(),
+            initial_store_size,
+            store_file,
+        }
+    }
+
+    /// Track a new insertion to the metrics, with the insertion time and size of the block.
+    pub fn add_insertion(&mut self, insertion_time: Duration, block_size: usize) {
+        self.insertion_time_per_block.push(insertion_time);
+        self.bytes_per_block.push(block_size);
+        self.num_insertions += 1;
+    }
+
+    /// Track the size of the store file.
+    pub fn record_store_size(&mut self) {
+        self.store_file_sizes.push(Self::get_store_size(&self.store_file));
+    }
+
+    /// Track the time it takes to query the block inputs.
+    pub fn add_get_block_inputs(&mut self, query_time: Duration) {
+        self.get_block_inputs_time_per_block.push(query_time);
+    }
+
+    /// Track the time it takes to query the batch inputs.
+    pub fn add_get_batch_inputs(&mut self, query_time: Duration) {
+        self.get_batch_inputs_time_per_block.push(query_time);
+    }
+
+    /// Get the size of the store file and its WAL file.
+    fn get_store_size(dump_file: &Path) -> u64 {
+        let store_file_size = std::fs::metadata(dump_file).unwrap().len();
+        let wal_file_size =
+            std::fs::metadata(format!("{}-wal", dump_file.to_str().unwrap())).unwrap().len();
+        store_file_size + wal_file_size
+    }
```

Comment:
nit: since this does not use `self` - it could probably be just a "pure" helper function.